lib/traject_plus/macros.rb

```diff
@@ -15,25 +15,60 @@ def transform_values(context, hash)
     # try a bunch of macros and short-circuit after one returns values
     def first(*macros)
       lambda do |record, accumulator, context|
-        macros.lazy.map do |lambda|
-          lambda.call(record, accumulator, context)
+        macros.lazy.map do |block|
+          block.call(record, accumulator, context)
         end.reject(&:blank?).first
       end
     end
 
+    def accumulate(&block)
+      lambda do |record, accumulator, context|
+        Array(block.call(record, context)).each do |v|
+          accumulator << v if v.present?
+        end
+      end
+    end
+
     # only accumulate values if a condition is met
-    def conditional(condition, lambda)
+    def conditional(condition, block)
       lambda do |record, accumulator, context|
         if condition.call(record, context)
-          lambda.call(record, accumulator, context)
+          block.call(record, accumulator, context)
         end
       end
     end
 
     def from_settings(field)
-      lambda { |_record, accumulator, context|
-        accumulator << context.settings.fetch(field)
-      }
+      accumulate do |record, context|
+        context.settings.fetch(field)
+      end
+    end
+
+    def copy(field)
+      accumulate do |_record, context|
+        Array(context.output_hash[field])
+      end
+    end
+
+    # apply the same mapping to multiple fields
+    def to_fields(fields, mapping_method)
```

Comment:
i know we added this originally in DLME, but I remained a bit surprised this isn't in core Traject somewhere. Perhaps a question issue to explore that.